cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -1275,7 +1307,24 @@ private ProgramSpecification getProgramSpecificationWithoutAuthz(ProgramId progr
     if (appSpec == null) {
       return null;
     }
-    return getExistingAppProgramSpecification(appSpec, programId);
+    return getExistingAppProgramSpecification(appSpec, programId.getProgramReference());
+  }
+
+  /**
+   * Returns the {@link ProgramSpecification} for the specified {@link ProgramReference programReference}
+   * without performing authorization enforcement.
+   *
+   * @param programReference the {@link ProgramReference program} for which the {@link ProgramSpecification}
+   * is requested
+   * @return the {@link ProgramSpecification} for the specified {@link ProgramId program}
+   */
+  @Nullable
+  private ProgramSpecification getLatestProgramSpecificationWithoutAuthz(ProgramReference programReference) {
```

Comment:
'Authz' ?

Reply:
short for "authorization", trying to follow the naming convention with `getProgramSpecificationWithoutAuthz`